vignettes/lifecycle.Rmd

```diff
@@ -41,7 +41,11 @@ There are 4 __development__ stages.
 
 1. <img src="figures/lifecycle-questioning.svg" alt = "Questioning" style="vertical-align:middle" /> The author is no longer convinced that the feature is the optimal approach. However, there are no recommended alternatives yet.
 
-Once the decision of discontinuing a feature has been made, it goes through 3 __deprecation__ stages.
+When a feature is no longer under development, it can be either __retired__ or __deprecated__.
```

Comment:
This doesn't feel quite right because I wouldn't say an stable function is under development either.

Do you want me to redraw the package lifecycle diagram for function components? I think having an overview picture would help.

Reply:
That'd be great!

Reply:
I think of stable functions as in development because users can request new features.

Reply:
I think that will confuse people — I think we should be framing stable as the probable end state for most functions; it's only occasionally that functions will be retired or deprecated.

Reply:
```suggestion
Once the decision of discontinuing a feature has been made, it can be either __retired__ or __deprecated__.
```

Reply:
A retired feature is not discontinued, only the development is discontinued.

Reply:
Oh, right. Can we say there's three development stages that sometimes lead to stability, which isn't primarily a development stage, and that stable functions are occasionally retired?

Reply:
I like the idea of making it clearer that retired functions are not discontinued. Maybe the stage should be `stable-retired`?

This comes up often, I had to mention on SO just 2 days ago that plyr wasn't deprecated.

Reply:
I'm suggesting to mention "stable" and "retired" in the same section in this document, which (I think) is similar in spirit to what you suggest? This would make "retired" a form of "stable", but I'm not really in love with the combo word.

What about ![immortal](https://img.shields.io/badge/lifecycle-immortal-gold.svg)?